### PR TITLE
Document DockerBuildImages properties

### PR DESCRIPTION
**Pull request by hugares**

### CODE REVIEW

**Comment by bmuschko** [comment]
Maybe I wasn't clear in https://github.com/bmuschko/gradle-docker-plugin/pull/823/files#r291587712 but what I meant was "document why we do not assign `final` and what the `labels` field does". I didn't necessarily propose to document all fields in this class though I doesn't hurt.

**Comment by hugares** [comment]
> Maybe I wasn't clear in https://github.com/bmuschko/gradle-docker-plugin/pull/823/files#r291587712 but what I meant was "document why we do not assign final and what the labels field does". I didn't necessarily propose to document all fields in this class though it doesn't hurt.

I know you did not suggest that but instead of documenting only the label property(including reason why it is not final), I thought it would be good to document them all and in my other PR, I will add the section to explain why the label property is not final.

**Comment by bmuschko** [comment]
@hugares Great, thanks! Much needed work.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy [hunk 1]
```diff
@@ -64,34 +64,58 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
     @Optional
     final SetProperty<String> tags = project.objects.setProperty(String)
 
+    /**
+     * When <code>true</code>, do not use docker cache when building the image.
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy]: The option would be to use `{@code}`. Would you mind using that instead?

Comment (hugares) [reply src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy]: Done

**Pull request merged**